### ISSUE TITLE
[Question] Middleware to use after @Authenticated()

### ISSUE THREAD

**Issue by zoubarevm** [Romakita/ts-express-decorators#39]
Hey!

I was wondering if there is a way to use middleware within the controller after an `@Authenticated` call. Seems like if I call a `@Use`, the middleware executes **before** the `$onAuth` call (I set a `request.user` within that call). 

Thanks in advance for your help!

**Comment by Romakita** [comment]
Hi @zoubarevm ,

Normally the order of the annotations defines the calling order of the middlewares. (Except for the @Get()/@Post() etc..., they are always the priority whatever order).

So If you make this:
```typescript
@Get('/')
@Authenticated()
@Use(myMiddleware)
myMethod(){

}

// With express.js it's equivalent to
app.get('/', $onAuth, myMiddleware, myMethod);
```

If this is not the case then there is a bug :)

See you

**Comment by zoubarevm** [comment]
Hi @Romakita ,

It seems like the `@Use` middleware is overwriting the `@Authenticated` call.

Here is what I have in my controller:

```javascript
@Get('/my/:id(\\d+)')
@Authenticated()
@Use(MyController.checkPermission('retrieve'))
public getById(@Request() request: Express.Request,
                    @PathParams('id') id: number) {
    ...
}
```

The `checkPermission` method is hit, but the `$onAuth` is not.

When I remove the `@Use`, the `$onAuth` is hit.

Any help is appreciated!

**Comment by Romakita** [comment]
Hello @zoubarevm ,

After code review, it's a bug. Typescript reverses the execution order of the decorators. I didn't know that XD.
 I'll provide a quick fix ;)

See you.

**Comment by Romakita** [comment]
Fixed :) v1.3.3

**Comment by zoubarevm** [comment]
@Romakita thanks for your hard work, but it seems like I'm getting an error on the latest build: 

```
HTTP Server error TypeError: Cannot read property 'forEach' of undefined
[1]     at Function.InjectorService.construct (/Projects/expressts2/node_modules/ts-express-decorators/lib/services/injector.js:45:13)
[1]     at /Projects/expressts2/node_modules/ts-express-decorators/src/services/injector.ts:108:37
```

<img width="265" alt="7ba7e7864b8f44fe8d22a85f730a809a 1" src="https://cloud.githubusercontent.com/assets/6774855/22351046/33d7e144-e3e4-11e6-9f9d-00d968b9014d.png">

**Comment by Romakita** [comment]
Hi @zoubarevm,

The injectorService isn't ready on this step. But we haven't a hook for that except the $onReady.

I'll tried to move the injectorService.load() to have the services loaded before the $onMountingMiddlewares hook.

**Comment by Romakita** [comment]
"One thing to note: I am calling the InjectorService in the mounting of middlewares for another middleware that I'm activating globally."

Ok. I'll move the InjectorService.load() to have all service ready before $onMountingMiddleware and update the documentation.

**Comment by Romakita** [comment]
Fixed v1.3.4

**Comment by zoubarevm** [comment]
@Romakita I use one of the injected services within the middleware. The latest version looks good :)

Thanks so much for your help!
